src/debuggability.js

```diff
@@ -141,6 +141,69 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+var supportedEvents = ["created", "chained", "fulfilled",
+    "rejected", "resolved", "cancelled"];
+
+Promise.on = function (eventName, hookFunction, simpleEventApi) {
```

Comment:
Just one API is sufficient. It will make this easier to use and understand and it should reduce some internal code as well I hope (currently the hooking stuff is very bizarre and complex).